### PR TITLE
Update SegmentationModel C++

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: tests/cpp/accuracy/test_accuracy.cpp [hunk 1]
```diff
@@ -122,8 +128,43 @@ TEST_P(ModelParameterizedTest, AccuracyTest)
...
+        auto model = SegmentationModel::create_model(DATA_DIR + "/" + modelPath);
+
+        for (size_t i = 0; i < modelData.testData.size(); i++) {
+            auto imagePath = DATA_DIR + "/" + IMAGE_PATH + "/" + modelData.testData[i].image;
+
+            cv::Mat image = cv::imread(imagePath);
+            if (!image.data) {
+                throw std::runtime_error{"Failed to read the image"};
+            }
+            cv::Mat predicted_mask[] = {model->infer(image)->resultImage};
```
Comment (AlexKoff88) [inline tests/cpp/accuracy/test_accuracy.cpp:141]: Based on the recent changes in the validation of the Python part. Don't you need to take min/max from the `predicted_mask` prior to defining the ranges for the histogram?

File: model_api/cpp/models/src/segmentation_model.cpp [hunk 2]
```diff
@@ -32,8 +32,45 @@
...
 #include "models/results.h"
 
-SegmentationModel::SegmentationModel(const std::string& modelFile, bool useAutoResize, const std::string& layout)
-    : ImageModel(modelFile, "standard", useAutoResize, layout) {}
+std::unique_ptr<SegmentationModel> SegmentationModel::create_model(const std::string& modelFile, const ov::AnyMap& configuration) {
+    auto core = ov::Core();
+    std::shared_ptr<ov::Model> model = core.read_model(modelFile);
+
+    // Check model_type in the rt_info, ignore configuration
+    std::string model_type = "segmentation";
```
[the remark this answers is not in the document]
Comment (Wovchena) [reply model_api/cpp/models/src/segmentation_model.cpp:40]: It was written in IR with lower latter. I will align it in OMZ

[the remark this answers is not in the document]
Comment (AlexKoff88) [reply model_api/cpp/models/src/segmentation_model.cpp:40]: Ok. Please consider doing this. I ok with either option as long as it looks the same across the types.

File: tests/cpp/accuracy/test_accuracy.cpp (hunk 1 above)
Comment (AlexKoff88) [reply tests/cpp/accuracy/test_accuracy.cpp:141]: Ok. I see that this is true according to the postprocessing defined in the segmentation model_type.

**Pull request merged**